src/Conversion/AutoConversion/AutoConvertStream.tsx

```diff
@@ -47,6 +51,11 @@ class AutoConvertStream extends LocalStorageComponent<Props, State, SavedState>
             <Button onClick={(event: React.MouseEvent<Button>) => this.onClearClick(event)}>Clear</Button>
           </ButtonGroup>
         </ButtonToolbar>
+        <input 
+          type="checkbox"
+          onClick={(event: React.MouseEvent<HTMLInputElement>) => this.toggleHomogenous(event)}
```

Comment:
You'll want to bind `this.state.homogeneous` to the `checked` property of the input element.  Then the state will be restored correctly.

Reply:
Maybe I'm doing it wrong, but I'm getting:

"Binds are forbidden in JSX attributes due to their rendering performance impact (jsx-no-bind)" - TSLint

Reply:
It should just be like:

```
<input checked={this.state.homogeneous} />
```

Reply:
You may also want to take a look at my semaphore UI usage of checkboxes, I listen to onChange and get the value of checked.